nexus/src/app/sagas/instance_create.rs

```diff
@@ -1310,58 +1365,72 @@ async fn sic_instance_ensure(
     } else {
         osagactx
             .nexus()
-            .instance_set_runtime(
-                &opctx,
-                &authz_instance,
-                &db_instance,
-                InstanceStateRequested::Running,
-            )
+            .instance_ensure_registered(&opctx, &authz_instance, &db_instance)
             .await
             .map_err(ActionError::action_failed)?;
     }
 
     Ok(())
 }
 
-async fn sic_noop(_sagactx: NexusActionContext) -> Result<(), ActionError> {
-    Ok(())
-}
-
-/// Ensure that the necessary v2p mappings exist for this instance
-async fn sic_v2p_ensure(
+async fn sic_instance_ensure_registered_undo(
     sagactx: NexusActionContext,
-) -> Result<(), ActionError> {
+) -> Result<(), anyhow::Error> {
     let osagactx = sagactx.user_data();
     let params = sagactx.saga_params::<Params>()?;
+    let datastore = osagactx.datastore();
+    let instance_id = sagactx.lookup::<Uuid>("instance_id")?;
     let opctx = crate::context::op_context_for_saga_action(
         &sagactx,
         &params.serialized_authn,
     );
-    let instance_id = sagactx.lookup::<Uuid>("instance_id")?;
+
+    let (.., authz_instance, db_instance) = LookupPath::new(&opctx, &datastore)
+        .instance_id(instance_id)
+        .fetch()
+        .await
+        .map_err(ActionError::action_failed)?;
 
     osagactx
         .nexus()
-        .create_instance_v2p_mappings(&opctx, instance_id)
+        .instance_request_state(
+            &opctx,
+            &authz_instance,
+            &db_instance,
+            InstanceStateRequested::Destroyed,
```

Comment:
Seems a little odd the undo for `ensure_registered` is via `instance_request_state`.

Reply:
Ah, I see sean's similar comment in `instance_manager`.